test/e2e_node/perf_test.go

```diff
@@ -0,0 +1,332 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"context"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/api/types/filters"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	"github.com/docker/docker/api/types"
+	docker "github.com/docker/docker/client"
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	toolboxPodName          = "perf-toolbox-pod"
+	toolboxContainer        = "perf-toolbox"
+	graphingPodName         = "perf-graphing-pod"
+	workloadPodPrefix       = "workload-pod-"
+	workloadContainerPrefix = "workload-container-"
+)
+
+type Workload struct{}
+
```

Comment:
Currently I think this test only works on COS. I think we should add a comment here to make this clear. Also I think we should skip the test for other OS versions. You can do something like this: https://github.com/kubernetes/kubernetes/blob/release-1.11/test/e2e/scheduling/nvidia-gpus.go#L72 that we used to have for gpu tests.

 Please also add some comments to explain what this test is doing and instructions on how to run the test.